### PR TITLE
Utility functions for `PMap`

### PR DESCRIPTION
**Pull request by SeungheonOh**
Depending on #561

### CODE REVIEW

File: plutarch-extra/Plutarch/Extra/Map.hs [hunk 1]
```diff
@@ -0,0 +1,335 @@
+{-# LANGUAGE QuantifiedConstraints #-}
+
+{- |
+ Module: Plutarch.Extra.Map
+ Copyright: (C) Liqwid Labs 2022
+ License: Apache 2.0
```
Comment (L-as) [inline plutarch-extra/Plutarch/Extra/Map.hs:6]: ? I can't accept it with this in. The license must be MIT. In addition, we don't have any header like this anywhere, albeit I'd be OK with it being shortened to a 2 lines with just the original author and sponsor.

File: plutarch-extra/Plutarch/Extra/Map.hs [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+import qualified Plutarch.List as PList
+
+{- | If a value exists at the specified key, apply the function argument to it;
+ otherwise, do nothing.
+
+ This is necessarily linear in the size of the map performance-wise, as we
+ have to scan the entire map to find the key in the worst case.
+-}
+padjust ::
+    forall (k :: S -> Type) (v :: S -> Type) (s :: S).
```
Comment (L-as) [inline plutarch-extra/Plutarch/Extra/Map.hs:62]: PType

**Review by L-as — CHANGES_REQUESTED** [review]

**Comment by L-as** [comment]
I can't update this PR.

**Comment by SeungheonOh** [comment]
I'll make a new PR from my fork. Liqwid-Labs fork seems to have problem with permission

**Pull request closed without merging**